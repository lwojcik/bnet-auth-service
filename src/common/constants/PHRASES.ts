import { REDIS } from './environment';

export const PHRASES = {
  accessToken: {
    received: (token: string) => `Received access token: ${token}`,
  },
  battlenet: {
    usingClientId: (clientId: string) =>
      `Using Battle.net client id: ${clientId}`,
    usingClientSecret: (secret: string) =>
      `Using Battle.net client secret: ${secret}`,
    receivedAccessToken: (token: string) => `Received access token: ${token},`,
  },
  cache: {
    usingKey: (key: string) => `Using Redis key: ${key}`,
    usingTTL: (secs: number) => `Set TTL to ${secs} seconds`,
    accessKeyNotSaved: `${REDIS.enable} set to false - access key not saved`,
    cacheServiceDisabled: `${REDIS.enable} set to false - returning 'null' as access key`,
  },
};
