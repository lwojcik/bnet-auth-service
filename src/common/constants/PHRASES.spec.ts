import { PHRASES } from './PHRASES';

const fnsToTest = [
  {
    feature: 'accessToken',
    items: [
      {
        name: 'received',
        arg: 'receivedTestAccessToken',
      },
      {
        name: 'cachingAccessToken',
        arg: 'testAccessTokenToCache',
      },
    ],
  },
  {
    feature: 'battlenet',
    items: [
      {
        name: 'usingClientId',
        arg: 'testCliendId',
      },
      {
        name: 'usingClientSecret',
        arg: 'testClientSecret',
      },
      {
        name: 'receivedAccessToken',
        arg: 'receivedTestAccessToken',
      },
    ],
  },
  {
    feature: 'cache',
    items: [
      {
        name: 'usingKey',
        arg: 'testCacheKey',
      },
      {
        name: 'usingTTL',
        arg: 'testTTLinSecs',
      },
    ],
  },
];

describe('Phrases', () => {
  it('should be defined', () => {
    expect(PHRASES).toMatchSnapshot();
  });

  it('should format phrases correctly', () => {
    fnsToTest.forEach((testedFn) => {
      testedFn.items.forEach((item) => {
        expect(
          PHRASES[testedFn.feature][item.name](item.arg)
        ).toMatchSnapshot();
      });
    });
  });
});
