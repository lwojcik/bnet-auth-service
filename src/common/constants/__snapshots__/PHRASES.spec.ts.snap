// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phrases should be defined 1`] = `
Object {
  "accessToken": Object {
    "cachingAccessToken": [Function],
    "noCachedAccessToken": "Found no cached access token or \\"refresh\\" option was used",
    "readingFromCache": "Checking for cached access token",
    "received": [Function],
  },
  "battlenet": Object {
    "receivedAccessToken": [Function],
    "usingClientId": [Function],
    "usingClientSecret": [Function],
  },
  "cache": Object {
    "accessKeyNotSaved": "BAS_REDIS_ENABLE set to false - access key not saved",
    "cacheServiceDisabled": "BAS_REDIS_ENABLE set to false - returning 'null' as access key",
    "usingKey": [Function],
    "usingTTL": [Function],
  },
  "errors": Object {
    "accessToken": "Received access token error",
  },
}
`;

exports[`Phrases should format phrases correctly 1`] = `"Received access token: receivedTestAccessToken"`;

exports[`Phrases should format phrases correctly 2`] = `"Caching access token: testAccessTokenToCache"`;

exports[`Phrases should format phrases correctly 3`] = `"Using Battle.net client id: testCliendId"`;

exports[`Phrases should format phrases correctly 4`] = `"Using Battle.net client secret: testClientSecret"`;

exports[`Phrases should format phrases correctly 5`] = `"Received access token: receivedTestAccessToken,"`;

exports[`Phrases should format phrases correctly 6`] = `"Using Redis key: testCacheKey"`;

exports[`Phrases should format phrases correctly 7`] = `"Set TTL to testTTLinSecs seconds"`;
